### ISSUE TITLE
Jetpack Section: Restore

### ISSUE THREAD

**Issue by zwarm** [wordpress-mobile/WordPress-Android#13328]
Parent #13267 

<img width="1779" alt="Screen Shot" src="https://user-images.githubusercontent.com/506707/104459246-7c04ea00-557a-11eb-96eb-a830d82fe339.png">

**Design**
TBD

**Tasks**

- [x] Introduce restore request body to endpoint ([FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1761))
- [x] FluxC - Update endpoints to include message and current_entry for restore 
- [x] Replace rewind status service with restore use cases #13902
- [x] Hook wizard back into Activity Log List #13914
- [x] Fix dark theme issue with the progress row #13874
- [x] Add feature flag check to Activity Log Detail view to launch wizard #13979
- [x] Add error logging anywhere an exception is deliberately thrown #13910 
- [x] Implement error 3X retry on fetch within GetRestoreUseCase #13930 
- [x] Change "additional information" names to `footnote` to better describe the item #13912
- [x] On unchecking all checkbox items on the "Details" screen, we might want to disable the action button.  #13957
- [x] Clear DB on null Fetch from API to support edge case where we are caught between running and complete. FluxC [PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1860) and WPAndroid [PR](https://github.com/wordpress-mobile/WordPress-Android/pull/13944)
- [x] Remove the "Notify me" button from progress view Issue #14088 

#### Steps
- [x]  Restore Details (choose items to restore) [PR](https://github.com/wordpress-mobile/WordPress-Android/pull/13773) 
- [x]  Restore Warning PR (https://github.com/wordpress-mobile/WordPress-Android/pull/13775)
- [x]  Restore In Progress [PR](https://github.com/wordpress-mobile/WordPress-Android/pull/13778) 
- [x]  Restore Complete/Done [PR](https://github.com/wordpress-mobile/WordPress-Android/pull/13779)

#### Design
- [x]  Restore Details (choose items to backup)
- [x]  Restore Warning
- [x]  Restore In Progress
- [x]  Restore Complete/Done
- [x]  Update copy on all screens #13919 
- [x] Error view - Status vs. general layout (see iOS) #13919 

#### Fixes 
- [x] When I click on "Restore site" it navigates to Warning screen -> there is a noticeable animation glitch -> the cancel button appears after a delay and the confirmation button changes it's vertical position a bit - #13921 
- [x] There is a noticeable delay after clicking on the "Confirm restore site" on the warning screen, it takes +- 1-2 seconds before the apps navigates me to the progress screen.
#### Tracks
- [x] Add tracking #13947 

#### Finishing Touches

- [x] Check Accessibility
- [x] Check Small Devices
- [x] Check RTL languages
- [x] Design Review

# Low priority (nice to have)
- [x] Replace rewind status service with another simpler solution for activity log details screen @ParaskP7 #14117 
~~- [ ] Make sure timestamp is shown on restore progress row when triggered from the web ~~ -- Issue has been moved outside of the Jetpack Section project #14083

**Comment by ParaskP7** [comment]
After having investigated the general `Backup Download` and `Restore` solution, I have gathered some observations I wanted to share. I already discussed most of that with @Annmarie and @momozw. Here I just want to post a summary of all that so that we are all aware and can move forward without surprises.
1) 🏧 We can only have one restore running at a time. If triggered on the web, the app shouldn’t be able to trigger another one. And vice-versa.
1) 🏧 We can only have one backup download running at a time. If triggered on the web, the app shouldn’t be able to trigger another one. And vice-versa.
1) 🏧 We can have both, a restore and a backup download running at the same time. In that case, two progress rows should be visible on the activity log/backup screens, one for the restore and one for the backup. This also means that we can trigger a restore on the app and a backup download on the web. And vice-versa.
1) ⚠️ Currently, when restore is triggered on the web, while the user has the activity log or backup screen opened, the restore progress row is not shown automatically unless a pull-to-refresh is triggered or the screens are re-launched. This makes it possible for restore to be triggered twice, or different restores to be triggered at the same time.
1) ⚠️ The same will apply to backup download later on.
1) ⚙️ On Android, when a restore is triggered then the ellipsis menu item is hidden from the rewindable rows. However, this should be updated to the way the iOS is handling this. On iOS, the ellipsis menu item is NOT hidden. Instead, the restore menu item action is being disabled. On Android, this behavior can cause problems as it does not allow triggering a backup download to run in parallel with a restore. And vice-versa, for when a backup download progress row will get implemented. The web is also handling this the same way with iOS.
1) ⚙️ On Android, @Annmarie has already implemented a mechanism to disallow triggering twice the same restore, on the restore detail screen. On iOS, you might want to consider doing the same. Then, when 6. gets implemented on Android, the same restriction should be applied to the menu item action, when that is not being disabled (due of 4.).
1) ⚙️ The same should apply to backup download later on.

**UPDATE**: On the Android side, after confirming with the team, `4 + 5` are a bit tricky to deal with and we will only be focusing on `6 + 7, 8` if we get the time.
